powersimdata/design/investment/create_mapping_files.py

```diff
@@ -67,6 +68,29 @@ def _find_nearest(series, polygons, search_dist):
     points_in_regions = gpd.sjoin(left_df=left_df, right_df=right_df, op="intersects")
     points_in_regions["dist"] = 0
 
+    # Since polygons may overlap, there can be duplicated buses that we want to filter
+    duplicated = points_in_regions.loc[points_in_regions.index.duplicated(keep=False)]
+    to_drop = set()
+    for bus in set(duplicated["bus_id"]):
+        entries = duplicated.query("bus_id == @bus")
+        coords = entries["geometry"].iloc[0].coords[0]  # First duped entry, only point
+        regions = set(entries["name_abbr"])  # noqa: F841
+        candidates = points_in_regions.query(
+            "index not in @duplicated.index and name_abbr in @regions"
+        )
+        neighbor = candidates.apply(
+            lambda x: haversine((x.geometry.x, x.geometry.y), coords), axis=1
+        ).idxmin()
+        closest_region = candidates.loc[neighbor, "name_abbr"]  # noqa: F841
```

Comment:
If I'm understanding correctly, the logic here builds on the assumption that the `closest_region` must be one of the regions for the duplicated bus. Just curious what if that's not the case, i.e. we have two entries as following
```
bus_1 region A
bus_1 region B
```
and the closest neighbor, `bus_2` is in `region C`, i.e. no other buses in `region A` and `region B`, what should we do?

If we are safe in that situation, i.e. that will never happen. Would it be easier to simply drop all the duplications and reassign the regions of those entries by looking up the closest neighbor region directly instead of figuring out which regions should be dropped (as implemented currently)?

Reply:
My initial thinking of only looking at buses within the overlapping regions was to limit computation time. However, since the number of duplicate is fairly small, and this process is done only infrequently, computation time is probably not the main consideration.

Thinking a bit more, I could definitely draw a picture where a bus lands within two overlapping region shapes, but the closest neighbor bus is within a third region. I think the chances of this are fairly low though, since the bus would have to be within an intersection of two regions (0.01% incidence in this data set) and also in close vicinity to a three-region border. The assumption that one of the regions is correct seems reasonable enough to me.

Reply:
Yeah, regarding the second point, three region overlapping is one case. What I was thinking about previously is for buses within two overlapping regions, what if there is no bus within `S(region_a) - S(region_b)` or `S(region_b) - S(region_a)`, in such situation, we only have buses within the overlapping zone for the two regions, i.e. candidates list is empty in l.78. However, I agree, certain possibility is low and probably we can ignore that.

Given the assumption that we can always find a nearest neighbor in either region for any bus within overlapping zones, what I was proposing is (still only looking at those duplicated buses) we can simply reassign region for those buses based on the nearest neighbor instead of figuring out which region should drop, i.e. 
```python
duplicated_flag = points_in_regions.index.duplicated(keep=False)
non_duplicated = points_in_regions.loc[~duplicated_flag]
duplicated = points_in_regions.loc[~points_in_regions.index.duplicated(keep='first')]
# simply look at all regions with duplicated buses here, sacrifice some computational time here comparing with only look at duplicated regions for corresponding buses
regions = points_in_regions.loc[duplicated_flag, "name_abbr"].unique()
candidates = non_duplicated.query("name_abbr in @regions")
duplicated["name_abbr"] = duplicated.apply(lambda x: candidates.iloc[find_closest_neighbor(x.geometry, candidates.geometry)]["name_abbr"])
points_in_regions = non_duplicated.append(duplicated)
```

Reply:
Quick comment on this resolved thread: I think we are good to go with the current implementations, just clarifying the corner cases that we don't cover and providing an alternative of implementation here.